### PR TITLE
Solo farmanp

### PR DESCRIPTION
**Pull request by farmanp**
@case-eee How does this FixedArray challenge look? I thought I figured it out but I didn't know if it would be this straightforward. I didn't know how else to test it besides asking an instructor who is familiar with the concept.

### CODE REVIEW

**Comment by farmanp** [comment]
@mattbaker How does this FixedArray challenge look? I thought I figured it out but I didn't know if it would be this straightforward. I didn't know how else to test it besides asking an instructor who is familiar with the concept.

File: source/fixed_array.rb [hunk 1]
```diff
@@ -1,3 +1,24 @@
...
+  attr_accessor :size, :array
 
+  def initialize(size, array)
+    @size = size
+    @array = Array.new(size, array)
+  end
+
+  def get(index)
+    if index <= self.size
+      self.array[index]
```
Comment (mattbaker) [inline source/fixed_array.rb]: Does `@array` need to be public? Could this be a private accessor?

Part of the point of FixedArray is to restrict you, so I'd be inclined to say no one outside the object should be able to access the ruby array underneath.

File: source/fixed_array.rb [hunk 2]
```diff
@@ -1,3 +1,24 @@
 class FixedArray
+  attr_accessor :size, :array
 
+  def initialize(size, array)
+    @size = size
+    @array = Array.new(size, array)
+  end
+
+  def get(index)
+    if index <= self.size
```
Comment (mattbaker) [inline source/fixed_array.rb]: `< self.size` — if the size of the array is 3 then you'll have array[0], array[1], and array[2]. array[3] would be invalid.

Add another test to demonstrate that `get` can't access something outside bounds (negative or >= size).

File: source/fixed_array.rb [hunk 3]
```diff
@@ -1,3 +1,24 @@
...
+      self.array[index]
+    else
+      raise "Index does not exist"
+    end
+  end
+
+  def set(index, element)
+    if index <= self.size
```
Comment (mattbaker) [inline source/fixed_array.rb:19]: ditto

**Comment by mattbaker** [comment]
@farmanp — see comments. Implementation is good but there are a couple things to fix before you move on

**Comment by farmanp** [comment]
@mattbaker  made a few more adjustments. It's more functional as a FixedArray but the only downside was that when I made a private accessor for the array, things weren't working anymore. I'll have to revisit but the small fixes are pretty solid.

File: source/fixed_array.rb [hunk 4]
```diff
@@ -1,3 +1,25 @@
 class FixedArray
+  attr_accessor :array
```
Comment (mattbaker) [inline source/fixed_array.rb:2]: Make it private! If you think you need it you need to re-think how you're using FixedArray

**Comment by mattbaker** [comment]
Make the `array` accessor private, otherwise this looks good

File: source/fixed_array.rb [hunk 5]
```diff
@@ -1,3 +1,25 @@
...
+  def get(index)
+    if index < self.size
+      @array[index]
+    else
+      raise "Index does not exist"
+   end
+  end
+
+  def set(index, element)
+    if index <= self.size
```
Comment (mattbaker) [inline source/fixed_array.rb:19]: Are you sure you want `<=`?

**Comment by mattbaker** [comment]
@farmanp — left ya some more comments, good improvements, couple more things to fix
